### ISSUE TITLE
Enhancement Request - Check Local Storage Available 

### ISSUE THREAD

**Issue by I-CRE8** [cferdinandi/form-saver#34]
Is it possible to add a hook to check if sufficient local storage space is available ?

**Comment by cferdinandi** [comment]
I'm intrigued by this. What's the desired use case here? In other words, how might you use this information?

For my future knowledge, I found this code which can be used to figure that out: https://glynrob.com/javascript/calculate-localstorage-space/

**Comment by I-CRE8** [comment]
It struck me that local storage is available to all and the limit is not just a limit for your application but for everyone else as well so if my forms depend on local storage to maintain their state as the user makes backwards and forwards if local storage is full then it will throw an error which potentially brings the whole process to a grinding halt so my thinking is that if local storage is full then I could use a callback to warn the user or even switch to a different method of saving the form state.

**Comment by cferdinandi** [comment]
I like that option... if full (or if there's not enough space relative to the data you're trying to save), run a callback and pass in the form data so you as the developer can figure out how you'd like to handle.

**Comment by I-CRE8** [comment]
Yes something along those lines, I have to confess I am a little confused about the limits of storage available as I have seen sources that say xMb total for everybody and some say xMB each user

**Comment by cferdinandi** [comment]
Still no time to work on this yet, but... I think the better approach is to alert when storage is full...

```js
try {
  localStorage.setItem("name", "Hello World!"); //saves to the database, "key", "value"
} catch (e) {
  if (e == QUOTA_EXCEEDED_ERR) {
    alert('Quota exceeded!'); //data wasn't successfully saved due to quota exceed so throw an error
  }
}
```

**Comment by I-CRE8** [comment]
Thanks for the reply, I will be starting work on a new project in the near future where I am going to try this. When I do I will let you know how I get on.
